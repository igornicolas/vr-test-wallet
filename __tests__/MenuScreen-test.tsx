import MenuScreen from '@/src/screens/Menu';
import { render } from '@testing-library/react-native';

describe('<MenuScreen />', () => {
    it('check if  text renders correctly on HomeScreen', () => {
        //@ts-ignore
        const { getByText } = render(<MenuScreen />);

        const title = getByText('Wallet Test');
        expect(title).toBeTruthy();
    });
    it('check if buttons renders correctly on HomeScreen', () => {
        //@ts-ignore
        const { getByText } = render(<MenuScreen />);

        const buttonGoToMyCards = getByText('meus cartões');
        const buttonGoToCreateCards = getByText('cadastrar cartão');

        expect(buttonGoToMyCards).toBeTruthy();
        expect(buttonGoToCreateCards).toBeTruthy();
    });
    it('check menuscreen renders correctly', () => {
        //@ts-ignore
        const { toJSON } = render(<MenuScreen />);
        const tree = toJSON();
        expect(tree).toMatchSnapshot();
    });
});
