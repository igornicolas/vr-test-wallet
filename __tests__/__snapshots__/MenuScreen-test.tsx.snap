// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuScreen /> check menuscreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "gap": 20,
      "justifyContent": "center",
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#3F3F3F",
          "fontFamily": "PTSansCaption_400Regular",
          "fontWeight": "400",
        },
        {
          "fontSize": 28,
          "lineHeight": 32,
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Wallet Test
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#12C2E9",
        "borderRadius": 12,
        "height": 55,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#3F3F3F",
            "fontFamily": "PTSansCaption_400Regular",
            "fontWeight": "400",
          },
          {
            "fontFamily": "PTSans_400Regular",
            "fontSize": 18,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      meus cartões
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#A5FF32",
        "borderRadius": 12,
        "height": 55,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#3F3F3F",
            "fontFamily": "PTSansCaption_400Regular",
            "fontWeight": "400",
          },
          {
            "fontFamily": "PTSans_400Regular",
            "fontSize": 18,
            "lineHeight": 20,
          },
          {
            "color": "#142995",
          },
        ]
      }
    >
      cadastrar cartão
    </Text>
  </View>
</View>
`;
